### PR TITLE
Adding CI - Flake8

### LINKED ISSUE #14: Adding CI using Flake8
Requirements:

  1. Setup Flake8 to manage and verify that the code we add conforms to proper Python syntax and coding standards.
  2. Add GitHub actions configuration to run flake8 on the code as we submit new PRs.
  fixes #13

### PR DESCRIPTION
**Pull request by Yardengz**
Add GitHub actions configuration
set GitHub to run Flake8 on the code as we submit new PRs
fixes #14

### CODE REVIEW

**Review by Davidco94 — APPROVED** [review]
LGTM

**Review by idan1356 — APPROVED** [review]
LGTM

**Review by shirazyomtov — APPROVED** [review]
LGTM

File: .github/workflows/flake8.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    container: fedora:34
+    steps:
+      - name: Install Pipenv and Git
+        run: dnf install -y pipenv git
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Setup environment
+        run: pipenv sync --dev
+      - name: Run Flake8
+        run: pipenv run flake8 --max-line-length 120
```
Comment (natifridman) [inline .github/workflows/flake8.yml:16]: Missing newline at end of file

**Review by natifridman — CHANGES_REQUESTED** [review]
Overall looks good. Added one comment

File: .github/workflows/flake8.yml (hunk 1 above)
Comment (Yardengz) [reply .github/workflows/flake8.yml]: fixed, please confirm.

Comment (Yardengz) [reply .github/workflows/flake8.yml]: fixed, please confirm.

**Review by natifridman — CHANGES_REQUESTED** [review]

File: .github/workflows/flake8.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
```
[the remark this answers is not in the document]
Comment (Yardengz) [reply .github/workflows/flake8.yml]: i fixed this.

**Review by natifridman — APPROVED** [review]
LGTM. Please squash the commits so we can merge it

**Review by natifridman — CHANGES_REQUESTED** [review]
Looks like the flake8 installation was removed from this PR

File: .github/workflows/flake8.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
```
[the remark this answers is not in the document]
Comment (Yardengz) [reply .github/workflows/flake8.yml]: Fixed, please approve

**Review by natifridman — APPROVED** [review]
LGTM

**Pull request merged**